libs/x-content/src/test/java/org/elasticsearch/xcontent/DotExpandingXContentParserTests.java

```diff
@@ -166,4 +166,64 @@ public void testNestedExpansions() throws IOException {
             {"first.dot":{"second.dot":"value","third":"value"},"nodots":"value"}\
             """);
     }
+
+    public void testGetTokenLocation() throws IOException {
```

Comment:
I meant to add a test to reproduce this issue when indexing a document, but I did not find an easy way to do so. It turns out that a lot of errors that we throw as MapperParsingException don't actually hold the xcontent location, and the ones that do are parse exception coming from jackson directly. For the latter, the token location is not aligned only while expanding start or end object, which makes it hard to recreate the situation where the wrong token location would get returned without my fix.

All this said, I came to reconsider how important this fix is. Maybe in practice nobody will ever notice...?

Reply:
Or maybe we should be including xcontent location in more parsing exceptions? It's a bit odd that we use the same exception type for parsing mappings and parsing documents as well.

Reply:
I opened #85083